README.md

```diff
@@ -1703,3 +1731,100 @@ will return this response:
+    }
+}
+```
+
+| Key            | Value Type     | Value description                                                        |
+|----------------|----------------|--------------------------------------------------------------------------|
+| proxy          | JSON Object    | An object with the store names as keys and their configuration as values |
+| <store_name>   | JSON Object    | An object with the store configuration                                   |
+| mode           | string         | The mode the store runs on (see below)                                   |
+
+Store modes:
+
+| Value        | Description                                                                            |
+|--------------|----------------------------------------------------------------------------------------|
+| passthru     |  The request to the proxy store will be stored as it is without any further processing |
+
+
+### Write to proxy store
+
+Perform a request against the http server path `/proxywrite/<store_name>`.
+Where `<store_name>` is a key (store name) inside the `proxy` object in the configuration.
+The expected response will have either `200` status code and the used offset as body or another status and an error body.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxywrite/test",
+    "method": "POST",
+    "query_params": {
+        "some": "param"
+    },
+    "header": {
+        "X-My-Header": 0
+    },
+    "body": {
+        "post": {
+            "my": ["body", "here"]
+        }
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+    "statuscode": 200,
+    "body": {
+        "offset": 0
+    }
+}
+```
+
+### Read from proxy store
+
+Whatever request performed against the server path `/proxyread/<store_name>?offset=<offset>`.
+Where:
+- `<store_name>` is a key inside the `proxy` object in the server configuration, aka the proxy store name
+- `<offset>` represents the entry to be retrieved in the proxy store requests collection. If not provided, 0 is assumed.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxyread/test",
+    "method": "GET",
+    "query_params": {
+        "offset": 0
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+    "header": { // Merged headers. original request headers prefixed with 'X-Request`
+        "X-Apitest-Proxy-Request-Method": ["POST"], // The method of the request to the proxy store
+        "X-Apitest-Proxy-Request-Path": ["/proxywrite/test?is=here"], // The url path requested (including query string)
+        "X-Apitest-Proxy-Request-Query": ["is=here&my=data_{{ $offset }}&some=value"], // The request query string only
+        "X-My-Header": ["blah"], // Original request custom header
+        "X-Apitest-Proxy-Store-Count": ["7"], // The number of requests stored
+        "X-Apitest-Proxy-Store-Next-Offset": ["1"] // The next offset in the store
+        ... // All other standard headers sent with the original request (like Content-Type)
```

Comment:
Why is that an Array? Since we return only one request, the array makes no longer sense.

Reply:
All headers (as query string parameters) can have multiple values per key, therefore are arrays in responses. 
You can see it in any expected response example in this exact document

Reply:
Makes no sense to me if the request only contains one item from the proxy. Why do I want an array?

Reply:
Yes, doesn't make sense, but as you cannot force to have a single header in a golang response struct, you'll always get an array there, even if you add only one (makes sense, headers can have multiple values). Therefore all apitests expect every single header to be an array. We could override this but it would mean to change base code and refactor a lot of tests. I would advise against it, if you're only interested in a single value, you can access the [0] index like we do in plenty of places for both headers and query strings